### PR TITLE
Sync TS typings with Flow typings

### PR DESCRIPTION
**Pull request by danielrearden**
This PR only changes the existing TS types.

### CODE REVIEW

**Comment by danielrearden** [comment]
@IvanGoncharov My current plan is to:

* Fix existing types (this PR)
* Break up the type declarations and move them into `/src` (this will also convert from using a namespace to ES module syntax)
* Add any missing type declarations
* Convert the tests to TS
* Convert the rest of the files to TS, extracting the Flow typings

File: types/index.d.ts [hunk 1]
```diff
@@ -56,64 +58,64 @@ declare namespace graphqlHTTP {
     /**
      * A value to pass as the context to the graphql() function.
      */
-    context?: unknown;
+    context?: Maybe<unknown>;
```
Comment (IvanGoncharov) [inline types/index.d.ts:61]: `unknown` should include `null` and `undefined` right?

File: types/index.d.ts [hunk 2]
```diff
@@ -56,64 +58,64 @@ declare namespace graphqlHTTP {
...
 
     /**
      * An object to pass as the rootValue to the graphql() function.
      */
-    rootValue?: unknown;
+    rootValue?: Maybe<unknown>;
```
Comment (IvanGoncharov) [inline types/index.d.ts:66]: same

File: types/index.d.ts (hunk 1 above)
Comment (IvanGoncharov) [reply types/index.d.ts:61]: I see it marked as `?mixed` in Flow.
So this change is ok 👍

File: types/index.d.ts (hunk 2 above)
Comment (IvanGoncharov) [reply types/index.d.ts:66]: 👍

File: types/index.d.ts (hunk 1 above)
Comment (IvanGoncharov) [reply types/index.d.ts:61]: I think it can fixed in separate PR both in Flow and TS

File: types/index.d.ts [hunk 3]
```diff
@@ -56,64 +58,64 @@ declare namespace graphqlHTTP {
...
 
     /**
      * A boolean to configure whether the output should be pretty-printed.
      */
-    pretty?: boolean | null;
+    pretty?: Maybe<boolean>;
```
Comment (IvanGoncharov) [inline types/index.d.ts:71]: One thing to note: after TS conversion we need to get rid of `Maybe` types, AFAIK they are not part of idiomatic TS and at that point would be just a Flow legacy.

**Comment by IvanGoncharov** [comment]
@danielrearden It looks like TS by default supports `undefined` as values of optional arguments and fields: https://www.typescriptlang.org/play/index.html#code/C4TwDgpgBA8mwEsD2A7AzlAvFA3gKCigDMB+ALijWACcEUBzAbjwF9m8BjVKqJeNAIwU4ibllwtO3YL34AmYfGTpxOYhQCuKACYQidCNqiSu6GX2BoAzItErsaohRQaANq+NSzsywBZbyhgO6lAA5ERISKHGQA

Moreover, it looks like Flow has the same behaviour:
https://flow.org/try/#0C4TwDgpgBA8mwEsD2A7AzlAvFA3gKCigDMB+ALijWACcEUBzAGjwF8BuPPAY1SqiXhoAjBTiJeWXC269g-QQCZR8ZOkk5iFAK4oAJhCJ0IuqNJ7o5A4GgAsy8WuwaiFAOREkSV6c5EdXB2IACipqckoaOnoASik8IiCdfUMUY2j4oPciV3SgA

**Comment by IvanGoncharov** [comment]
@danielrearden Based on the previous comment I cleaned up Flow typings in #624 
Can you please sync this PR?

> My current plan is to:

Sounds good 👍

**Pull request merged**